### PR TITLE
{Error Improvement} Error handling guidelines for extensions

### PR DESCRIPTION
**Pull request by houk-ms**
**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
@jiasli @evelyn-ys for awareness

**Review by dbradish-microsoft — COMMENTED** [review]
@houk-ms, I don't approve or request changes to PRs in this repo.  I defer this requested review to @jiasli.

File: doc/error_handling_guidelines.md [hunk 1]
```diff
@@ -4,11 +4,12 @@ This document aims to provide the guidelines for command group authors to onboar
...
 
-For the new commands authoring, here are what need to be done to get onboard. __Any PRs not following these rules here will be rejected.__ For the existing `CLIError` in command groups, command group owners should schedule to replace them with the new error types.
+For the new commands or features authoring in CLI modules, here are what need to be done to get onboard. __Any PRs not following these rules here will be rejected.__ For the existing `CLIError` in command groups, command group owners should schedule to replace them with the new error types.
 1. [__Mandatory__] Use the newly designed error types instead of `CLIError`. See section [Error Type](#Error-Type)
 2. [__Mandatory__] Follow the error message authoring guidelines. See section [Error Message](#Error-Message)
 3. [__Recommended__] Provide recommendations for users to take action. See section [Error Recommendation](#Error-Recommendation)
 
+__For CLI extensions, it's optional to adopt to the new error types. If applying the new error types,  `azext.minCliCoreVersion` should be set to `2.15.0` or higher versions.__
```
Comment (qwordy) [inline doc/error_handling_guidelines.md:12]:
```suggestion
__For CLI extensions, it's optional to adopt the new error types. If applying the new error types,  `azext.minCliCoreVersion` should be set to `2.15.0` or higher versions.__
```

Comment (qwordy) [reply doc/error_handling_guidelines.md:12]: Or "adapt to"

**Review by qwordy — APPROVED** [review]

**Pull request merged**